src/Orleans/Messaging/SqlMembershipTable.cs

```diff
@@ -21,489 +21,229 @@ MIT License
 TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */
 
+using Orleans.Messaging;
+using Orleans.Runtime.Configuration;
+using Orleans.Runtime.Storage.Relational;
 using System;
-using System.Net;
-using System.Text;
-using System.Data;
-using System.Data.SqlClient;
 using System.Collections.Generic;
 using System.Threading.Tasks;
 
-using Orleans.Messaging;
-using Orleans.Runtime.Configuration;
 
 namespace Orleans.Runtime.MembershipService
 {
-    internal class SqlMembershipTable : IMembershipTable, IGatewayListProvider
+    internal class SqlMembershipTable: IMembershipTable, IGatewayListProvider
     {
-        private string deploymentId;
-        private string connectionString;
+        private string deploymentId;        
         private TimeSpan maxStaleness;
+        private TraceLogger logger;
+        private IRelationalStorage database;
+
 
         public async Task InitializeMembershipTable(GlobalConfiguration config, bool tryInitTableVersion, TraceLogger traceLogger)
         {
+            logger = traceLogger;
             deploymentId = config.DeploymentId;
-            connectionString = config.DataConnectionString;
 
+            logger.Verbose3("SqlMembershipTable.InitializeMembershipTable called.");
+                                                
+            //TODO: Orleans does not yet provide the type of database used (to, e.g., to load dlls), so SQL Server is assumed.
+            database = RelationalStorageUtilities.CreateGenericStorageInstance(WellKnownRelationalInvariants.SqlServer, config.DataConnectionString);
```

Comment:
I'm thinking of opening an issue to word out the work needed in more detail. The very least that needs to be done is to transmit ``provider invariantname``, such as *System.Data.SqlClient* to the system. It could be a new attribute in Orleans configuration, but a little additional twist in the matter is that "normally" some of these are controller also in ``.config`` files in the ``system.data`` block. [MSDN Documentation](https://msdn.microsoft.com/en-us/library/dd0w4a2z%28v=vs.110%29.aspx), a fairly well explained [blog post](https://lukewickstead.wordpress.com/2014/02/26/ado-net-connections-and-data-providers/). I think this case warrants to think a solution that should work in any case.